### ISSUE TITLE
LeJEPA for Graphs

### ISSUE THREAD

**Issue by sebastian9991** [sebastian9991/TGFM#38]
I want to have some initial experiments on LeGraph, which defines using LeJEPA for graphs. 

- [x]  Data Preparation. We need to decide how _when_ we will construct the global and local views.  In particular, if we will be constructing them as we iterate over our subgraphs/graphs or if we will pre-construct them. 
- [x] Embed global/local view methods
- [x] LeJEPA model. Which is essentially GraphGPS + LeJEPA loss.
- [x] Overall training script. Perhaps some considerations on distributed training, when we consider this for large-scale pre-training.

**Comment by sebastian9991** [comment]
We will be using the data found: https://pytorch-geometric.readthedocs.io/en/2.5.3/modules/datasets.html
